front/app/components/Form/Components/Controls/MapInput/utils.tsx

```diff
@@ -0,0 +1,630 @@
+
+        // Remove any existing temporary graphics
+        temporaryDragGraphic.current &&
+          mapView.graphics.remove(temporaryDragGraphic.current);
+
+        // Create a temporary "preview" point graphic and add it to the map view
+        temporaryDragGraphic.current = pointBeingDragged?.current?.clone();
+
+        // Change the symbol colour so we can identify it as the preview point
+        temporaryDragGraphic.current.symbol = getShapeSymbol({
+          shape: 'circle',
+          color: tenantSecondaryColor,
+          sizeInPx: isMobileOrSmaller ? 26 : 20,
+          outlineColor: colors.white,
+          outlineWidth: 2,
+        });
+
+        // Generate temporary line graphics between the preview point and existing vertices
+        mapView &&
+          generateLinePreview({
+            mapView,
+            data,
+            inputType,
+            pointBeingDragged,
+            event,
+          });
+
+        // Add the preview graphic to the map
+        if (temporaryDragGraphic.current) {
+          temporaryDragGraphic.current.geometry = mapView.toMap(event);
+          mapView.graphics.add(temporaryDragGraphic.current);
+        }
+      }
+    } else if (event.action === 'end') {
+      // END ACTION: Update the form data with the new coordinates and remove temporary graphics
+
+      if (pointBeingDragged) {
+        event.stopPropagation();
+        if (temporaryDragGraphic.current) {
+          // Remove the temporary drag graphic
+          mapView.graphics.remove(temporaryDragGraphic.current);
+
+          // Get the original point we dragged and update its geometry (and save the updated data)
+          const dataCoordinates = getCoordinatesFromMultiPointData(
+            data,
+            inputType
+          );
+
+          const newData = dataCoordinates?.map((coordinates: number[][]) => {
+            const longitude = coordinates[0];
+            const latitude = coordinates[1];
+            if (
+              longitude ===
+                pointBeingDragged?.current?.geometry?.['longitude'] &&
+              latitude === pointBeingDragged?.current?.geometry?.['latitude']
+            ) {
+              // This is the original point the user tried to drag, so
+              // now we update the geometry.
+              return [
+                temporaryDragGraphic?.current?.geometry?.['longitude'],
+                temporaryDragGraphic?.current?.geometry?.['latitude'],
+              ];
+            } else {
+              return coordinates;
+            }
+          });
+
+          handleMultiPointChange && handleMultiPointChange(newData);
+
+          // Reset the variables and remove graphics
+          mapView.graphics.removeAll();
+          pointBeingDragged.current = null;
+          temporaryDragGraphic.current = null;
+        }
+      }
+    }
+  });
+};
+
+type HandlePointDragProps = {
+  mapView: MapView | null | undefined;
+  handleMultiPointChange:
```

Comment:
Maybe we could move the check for this `handleMultiPointChange` function to the `useEffect`? Because I suppose if it's undefined, all of this code shouldn't run anyway